crates/pure-stage/src/effect.rs

```diff
@@ -0,0 +1,264 @@
+use crate::{cast_msg, CallId, CallRef, Instant, Message, Name, StageRef};
+use std::{any::Any, fmt::Debug, time::Duration};
+use tokio::sync::oneshot;
+
+/// An effect emitted by a stage (in which case T is `Box<dyn Message>`) or an effect
+/// upon whose resumption the stage waits (in which case T is `()`).
+#[derive(Debug)]
+pub(crate) enum StageEffect<T> {
+    Receive,
+    Send(
+        Name,
+        T,
+        // this is present in case the send is the first part of a call effect
+        Option<(Duration, oneshot::Receiver<Box<dyn Message>>, CallId)>,
+    ),
```

Comment:
seems to me having named fields is clearer as it's probably the case we'll quickly forget exactly what those tuple values are

```suggestion
    Send {
        name: Name,
        msg: T,
        // this is present in case the send is the first part of a call effect
        call_param: Option<(Duration, oneshot::Receiver<Box<dyn Message>>, CallId)>,
    },

```